File: daily/Binary Search/74-Search-a-2D-Matrix.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        rows, cols = len(matrix), len(matrix[0])
        for r in range(rows):
            if matrix[r][cols - 1] >= target:
                left, right = 0, cols - 1
                while left <= right:
                    mid = (left + right) // 2
                    if matrix[r][mid] == target:
                        return True
                    elif matrix[r][mid] > target:
                        right = mid - 1
                    else:
                        left = mid + 1
        return False

File: daily/Binary Search/74-Search-a-2D-Matrix.go
package daily

func searchMatrix(matrix [][]int, target int) bool {
	rows, cols := len(matrix), len(matrix[0])
	for i := 0; i < rows; i++ {
		if matrix[i][cols-1] >= target {
			low, high := 0, cols-1
			for low <= high {
				mid := (low + high) / 2
				if matrix[i][mid] == target {
					return true
				} else if matrix[i][mid] > target {
					high = mid - 1
				} else {
					low = mid + 1
				}
			}
		}
	}
	return false
}
